internal/services/dataprotection/data_protection_backup_instance_kubernetes_cluster_resource.go

```diff
@@ -0,0 +1,354 @@
+	SnapshotResourceGroupName  string                       `tfschema:"snapshot_resource_group_name"`
+	BackupDatasourceParameters []BackupDatasourceParameters `tfschema:"backup_datasource_parameters"`
+}
+
+type BackupDatasourceParameters struct {
+	IncludedNamespaces          []string `tfschema:"included_namespaces"`
+	IncludedResourceTypes       []string `tfschema:"included_resource_types"`
+	ExcludedNamespaces          []string `tfschema:"excluded_namespaces"`
+	ExcludedResourceTypes       []string `tfschema:"excluded_resource_types"`
+	LabelSelectors              []string `tfschema:"label_selectors"`
+	VolumeSnapshotEnabled       bool     `tfschema:"volume_snapshot_enabled"`
+	ClusterScopeResourceEnabled bool     `tfschema:"cluster_scope_resource_enabled"`
+}
+
+type DataProtectionBackupInstanceKubernatesClusterResource struct{}
+
+var _ sdk.Resource = DataProtectionBackupInstanceKubernatesClusterResource{}
+
+func (r DataProtectionBackupInstanceKubernatesClusterResource) ResourceType() string {
+	return "azurerm_data_protection_backup_instance_kubernetes_cluster"
+}
+
+func (r DataProtectionBackupInstanceKubernatesClusterResource) ModelObject() interface{} {
+	return &BackupInstanceKubernatesClusterModel{}
+}
+
+func (r DataProtectionBackupInstanceKubernatesClusterResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return backupinstances.ValidateBackupInstanceID
+}
+
+func (r DataProtectionBackupInstanceKubernatesClusterResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"location": commonschema.Location(),
+
+		"vault_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: backupinstances.ValidateBackupVaultID,
+		},
+
+		"backup_policy_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: backuppolicies.ValidateBackupPolicyID,
+		},
+
+		"kubernetes_cluster_id": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateKubernetesClusterID,
+		},
+
+		"snapshot_resource_group_name": commonschema.ResourceGroupName(),
+
+		"backup_datasource_parameters": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"excluded_namespaces": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						ForceNew: true,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+					},
+					"excluded_resource_types": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						ForceNew: true,
+						Elem: &schema.Schema{
+							Type: schema.TypeString,
+						},
+					},
+					"cluster_scope_resource_enabled": {
```

Comment:
The naming of this property could use some fine-tuning to be more precise. Since it applies cluster wide to multiple resources I think it should be
```suggestion
					"cluster_scoped_resources_enabled": {
```

Reply:
Fixed.